pkg/serverutils/crdutils.go

```diff
@@ -0,0 +1,44 @@
+package serverutils
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// CrdResponseFilter shall filter partial metadata from knative event sources CRDs before propagating
+func CrdResponseFilter(w http.ResponseWriter, r io.Reader) {
+	var typedRes map[string]interface{}
+	err := json.NewDecoder(r).Decode(&typedRes)
+	if err != nil {
+		plog.Errorf("%s", err)
+		return 
+	}
+
+	filteredData := map[string]interface{}{
+		"kind":       typedRes["kind"],
+		"apiVersion": typedRes["apiVersion"],
+		"metadata":   typedRes["metadata"],
+	}
+
+	items := make([]interface{}, 0)
```

Comment:
Can we consider creating a struct with the fields we need instead of using `interface{}`?